File: backend/src/main/java/com/technicalescaperoom/backend/controller/player/PlayerRiddleController.java
package com.technicalescaperoom.backend.controller.player;

import com.technicalescaperoom.backend.config.security.PlayerPrincipal;
import com.technicalescaperoom.backend.dto.player.RiddleDto.RiddleBoardStateResponseDto;
import com.technicalescaperoom.backend.dto.player.RiddleDto.RiddleSubmissionRequest;
import com.technicalescaperoom.backend.dto.player.RiddleDto.RiddleSubmissionResponseDto;
import com.technicalescaperoom.backend.service.RiddleService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/player/riddles")
@RequiredArgsConstructor
public class PlayerRiddleController {

    private final RiddleService riddleService;

    @GetMapping
    public ResponseEntity<RiddleBoardStateResponseDto> getRiddleBoardState(
            @AuthenticationPrincipal PlayerPrincipal principal
    ) {
        RiddleBoardStateResponseDto response = riddleService.getRiddleBoardState(principal);
        return ResponseEntity.ok(response);
    }

    @PostMapping("/submit")
    public ResponseEntity<RiddleSubmissionResponseDto> submitRiddleAnswer(
            @AuthenticationPrincipal PlayerPrincipal principal,
            @Valid @RequestBody RiddleSubmissionRequest request
    ) {
        RiddleSubmissionResponseDto response = riddleService.submitRiddleAnswer(principal, request);
        return ResponseEntity.ok(response);
    }
}

File: backend/controllers/player/playerRiddleController.js
const riddleService = require('../../services/riddleService');

class PlayerRiddleController {
  async getRiddleBoardState(req, res, next) {
    try {
      const response = await riddleService.getRiddleBoardState(req.player);
      res.json(response);
    } catch (err) {
      next(err);
    }
  }

  async submitRiddleAnswer(req, res, next) {
    try {
      const response = await riddleService.submitRiddleAnswer(req.player, req.body);
      res.json(response);
    } catch (err) {
      next(err);
    }
  }
}

module.exports = new PlayerRiddleController();
